utils/sl/env_modules.py

```diff
@@ -21,100 +21,64 @@
 """Wrapper for Modules so we can load an MPI before builds or tests"""
 from __future__ import print_function
 import os
+import sys
 import subprocess
-try:
-    from subprocess import DEVNULL
-except ImportError:
-    DEVNULL = open(os.devnull, "wb")
+import errno
+from subprocess import PIPE, Popen
 from distutils.spawn import find_executable
-import re
 
-class _env_module():
+class _env_module(): # pylint: disable=invalid-name
     """Class for utilizing Modules component to load environment modules"""
     env_module_init = None
-    _mpi_map = {"mpich":['mpi/mpich-x86_64', 'mpich'],
+    _mpi_map = {"mpich":['mpi/mpich-x86_64', 'gnu-mpich'],
                 "openmpi":['mpi/openmpi3-x86_64', 'gnu-openmpi',
                            'mpi/openmpi-x86_64']}
 
     def __init__(self):
         """Load Modules for initializing envirables"""
-        self._module_func = self._init_module_func(lambda *x: False)
+        # Leap 15's lmod-lua doesn't include the usual module path
+        # in it's MODULEPATH, for some unknown reason
+        os.environ["MODULEPATH"] = ":".join([os.path.sep +
+                                             os.path.join("usr", "share",
+                                                          "modules"),
+                                             os.path.sep +
+                                             os.path.join("etc",
+                                                          "modulefiles")] +
+                                            os.environ.get("MODULEPATH",
+                                                           "").split(":"))
         self._module_load = self._init_mpi_module()
 
-    @staticmethod
-    def _module_old(*args):
-        """invoke module and save environment"""
-        cmd = ['/usr/bin/modulecmd', 'python']
+    def _module_func(self, command, *arguments): # pylint: disable=no-self-use
+        num_args = len(arguments)
+        cmd = ['/usr/share/lmod/lmod/libexec/lmod', 'python', command]
+        if num_args == 1:
+            cmd += arguments[0].split()
+        else:
+            cmd += list(arguments)
 
-        for arg in args:
-            cmd.append(arg)
-        output = subprocess.check_output(cmd)
-        # pylint: disable=exec-used
-        exec(output)  #nosec
-        # pylint: enable=exec-used
+        try:
+            proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
+        except OSError as error:
+            if error.errno == errno.ENOENT:
+                return None, None
+        except FileNotFoundError:
```

Comment:
(pylint-bad-except-order) Bad except clauses order (OSError is an ancestor class of FileNotFoundError)